### ISSUE TITLE
Improvements to search functionality in the website

### ISSUE THREAD

**Issue by ac10n** [ecadlabs/taquito#2858]
Currently, I observe these search related problems. These can affect a user trying to find information.
The information I'm trying to find is: `RemoteSigner`.

1. When going to the website, in home page, the search functionality returns nothing
2. In docs, in any version other than the currently latest version, the search functionality returns nothing
3. When in the current version, the most relevant result is not shown in the first page, and then when clicking on the "See all 11 results":
4. We are taken to a page with Dropbox for version, where the item is on "next", so the search returns nothing.

As a user, I could not guess changing the version fixes this.

Also, searching in Google for `RemoteSigner site:tezostaquito.io` does not return the "Class RemoteSigner" in the first page of results

**Comment by ac10n** [comment]
Interestingly, the current [sitemap](https://tezostaquito.io/sitemap.xml) file includes the old version, but not the current 19.x version.

```
<url>
<loc>https://tezostaquito.io/docs/17.5.0/tutorial_links</loc>
<changefreq>weekly</changefreq>
<priority>0.5</priority>
</url>
```

Also, it does include the [link to typedoc](https://tezostaquito.io/typedoc/).

**Comment by ac10n** [comment]
The current experience:

https://github.com/ecadlabs/taquito/assets/5188263/299fb65e-77a5-42fa-b160-33df5c8c25a5

**Comment by ac10n** [comment]
It seems to be a [known issue](https://github.com/facebook/docusaurus/issues/5084) that the search page does not respect the DocSearch options, so while turning the contextualSearch off changes the behaviour of the search modal, it does not affect the search page.

I swizzled the search page. In a separate commit, I modified it to respect the option.
After merging, and verifying that the search works properly, it might be fixed for the searchpage as well, because a re-index with `contextualSearch:off` will fix that, according to the linked issue.

In that case, we can do a separate PR to undo the swizzling of the SearchPage, which is a good idea, as it reduces the code we need to maintain.
